posthog/hogql_queries/lifecycle_query_runner.py

```diff
@@ -250,3 +251,25 @@ def periods_query(self):
                 timings=self.timings,
             )
         return periods_query
+
+    def _is_stale(self, cached_result_package):
+        date_to = self.query_date_range.date_to()
+        interval = self.query_date_range.interval_name
+        return is_stale(self.team, date_to, interval, cached_result_package)
+
+    def _refresh_frequency(self):
+        date_to = self.query_date_range.date_to()
+        date_from = self.query_date_range.date_from()
+        interval = self.query_date_range.interval_name
+
+        delta_days: Optional[int] = None
+        if date_from and date_to:
+            delta = date_to - date_from
+            delta_days = ceil(delta.total_seconds() / timedelta(days=1).total_seconds())
+
+        refresh_frequency = BASE_MINIMUM_INSIGHT_REFRESH_INTERVAL
+        if interval == "hour" or (delta_days is not None and delta_days <= 7):
+            # The interval is shorter for short-term insights
+            refresh_frequency = REDUCED_MINIMUM_INSIGHT_REFRESH_INTERVAL
+
+        return refresh_frequency
```

Comment:
This feels like something we could standardise across all queries into `query_runner.py`?

Reply:
Yep. There are a few special cases e.g. RetentionFilter uses `period` instead of `interval`, so I thought it's best to look at each query individually first and the unify the handling. Easy to forget this place otherwise.

Also, should we cache other HogQL queries e.g. those based on a date range?